sympy/core/symbol.py

```diff
@@ -577,6 +578,15 @@ def matches(self, expr, repl_dict=None, old=False):
 
 _range = _re.compile('([0-9]*:[0-9]+|[a-zA-Z]?:[a-zA-Z])')
 
+@overload
```

Comment:
I think it is time to start using those pyi-files or whatever they were called. This looks just horrendous... (And does NOT simplify for a developer to read the code etc.)

Reply:
I agree that this looks horrendous but maybe common sympy functions should not have such complicated type semantics in the first place. Given that the semantics actually are this complicated maybe it's good to make that clear with annotations.

Almost all use of `symbols` is just calling it with a single string and getting back either a single symbol or a tuple of symbols. I very rarely see anyone using any of the other overloads so I think it was a mistake to add them.

Reply:
And then it would "just" be:
`def symbols(names: str, *, cls=Symbol, **args) -> Union[Symbol, List[Symbol]]`
?

Reply:
Except that we should probably consistently rename `List` to `tList` as we have our own `List` class (and it will be a mess to remember which should be renamed and not, maybe have a renaming file in `utilities` so that one can import `tUnion` etc directly?).

Reply:
With PEP 585 you don't need `typing.Tuple` etc so as of Python 3.9 it's just:
```
Symbol | tuple[Symbol]
```
https://www.python.org/dev/peps/pep-0585/

Reply:
Actually the `|` syntax is Python 3.10:
https://www.python.org/dev/peps/pep-0604/